libs/native/src/main/java/org/elasticsearch/nativeaccess/lib/LoaderHelper.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.nativeaccess.lib;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * A utility for loading libraries from Elasticsearch's platform specific lib dir.
+ */
+public class LoaderHelper {
+    private static final Path platformLibDir = findPlatformLibDir();
+
+    private static Path findPlatformLibDir() {
+        // tests don't have an ES install, so the platform dir must be passed in explicitly
+        String path = System.getProperty("es.nativelibs.path");
+        if (path != null) {
+            return Paths.get(path);
+        }
+
+        Path platformDir = Paths.get("lib", "platform");
```

Comment:
Rather than implicitly relying on the working directory being the distribution directory, could we use the `ES_HOME` environment variable, or the `es.path.home` system property? Both of which are set by our CLI scripts. These _should_ always be the same, but presumably we pass those things to ES for a reason.

Reply:
`ES_HOME` only exists locally the shell (set in `bin/elasticsearch-env`, and the end if that script sets the current directory to it). It's true that we pass it through to the cli as es.path.home, but that's more historical, and we don't actually use that (we pass it through to the main process via the encoded node Settings object on startup).

IMO it's not implicit, it's just part of the contract with the shell script: the working directory of the CLI (and the main process) is the ES home dir.

Reply:
Also note we were already looking at the working dir for determining the platform dir before, see the changes in SystemJvmOptions.